modules/kubernetes-addons/prometheus/main.tf

```diff
@@ -75,4 +75,52 @@ resource "helm_release" "prometheus" {
     }
   }
 
+  depends_on = [kubernetes_namespace_v1.prometheus]
+
+}
+
+resource "kubernetes_namespace_v1" "prometheus" {
+  metadata {
+    name = local.helm_config["namespace"]
+    labels = {
+      "app.kubernetes.io/managed-by" = "terraform-ssp-amazon-eks"
+    }
+  }
+}
+
+# Amazon Managed Prometheus Resources
+
+resource "aws_prometheus_workspace" "amp_workspace" {
+  count = var.amazon_prometheus_workspace_id == null && var.enable_amp_for_prometheus ? 1 : 0
+  alias = format("%s-%s", "amp", var.eks_cluster_id)
+}
+
+module "irsa" {
+  for_each = { for k, v in local.irsa_config : k => v if var.enable_amp_for_prometheus }
```

Comment:
We can get rid of the for_each if we don't rely on the locals and just have two irsa blocks ;)

Reply:
I can change this to two IRSA blocks

Reply:
done